.github/workflows/ci.yaml

```diff
@@ -65,19 +65,19 @@ jobs:
       run: sudo chown -R $(whoami):$(id -ng) ~/.cargo/
 
     - name: Cache cargo registry
-      uses: actions/cache@v1
+      uses: actions/cache@v2
       with:
         path: ~/.cargo/registry
         key: ${{ runner.os }}-cargo-registry-${{ hashFiles('**/Cargo.lock') }}
 
     - name: Cache cargo index
-      uses: actions/cache@v1
+      uses: actions/cache@v2
       with:
         path: ~/.cargo/git
         key: ${{ runner.os }}-cargo-index-${{ hashFiles('**/Cargo.lock') }}
 
     - name: Cache cargo target dir
-      uses: actions/cache@v1
+      uses: actions/cache@v2
       with:
         path: target
         key: ${{ runner.os }}-cargo-build-target-${{ hashFiles('**/Cargo.lock') }}
```

Comment:
v2 allows specifying the list of paths, so we can merge all three entries into one I think.

Reply:
Yep, we could do that way! But the `target` dir size will be big sometimes so I'd recommend to separate its key and `.cargo` dir stuff, how about?

Reply:
Could elaborate a bit why that would be beneficial? I think I don’t understand something: we’ll still have to download/upload both, no?

We have https://github.com/rust-analyzer/rust-analyzer/blob/0f8af1b8246a31b1df60d3c60680954aefc9ab27/xtask/src/lib.rs#L123 to keep the size of target smaller, but that doesn’t sync with the list of crates, so it is bitrotted  atm :(

Reply:
Note that https://github.com/rust-analyzer/rust-analyzer/pull/5361 adds a couple more caches. I think #5361 doesn't need to cache `./target`, and that actually is a good reason to split .cargo and target into different keys!

Reply:
> Could elaborate a bit why that would be beneficial?

In the past, we often encountered cache size limit and unpacking issues so it was a workaround to reduce per-cache size. And, in v2, sometimes it takes less time to upload a large cache in two parts than it does to upload it in one go AFAIK.
Anyway, given #5361, I agree to separate them, I'll update it shortly.